### ISSUE TITLE
oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory

### ISSUE THREAD

**Issue by vitia** [docker/docker#22746]
**Output of `docker version`:**

```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 9
 Running: 7
 Paused: 0
 Stopped: 2
Images: 146
Server Version: 1.11.1
Storage Driver: devicemapper
 Pool Name: docker-253:2-814311901-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 62.84 GB
 Data Space Total: 107.4 GB
 Data Space Available: 44.53 GB
 Metadata Space Used: 32.6 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.115 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /home/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /home/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 3.10.0-327.10.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 11.58 GiB
Name: docker1
ID: D7L5:RJ5K:CG6A:UCPB:CHOR:ND5B:GRXU:IE7L:CFMZ:ABYW:6PFK:LICR
Docker Root Dir: /home/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**

```
#!/bin/bash -x
trap 'docker rmi oci_error ||:; docker rm -f oci_error ||:' EXIT
container_id=`docker run -d -v /tmp:/workspace --name oci_error centos:6 tail -f /dev/null`
docker exec --interactive=false $container_id date -u
image_id=`docker commit --pause=true $container_id oci_error`
docker exec --interactive=false $container_id date -u
```

**Describe the results you received:**

```
[root@docker1 ~]# ./oci_error.sh
+ trap 'docker rmi oci_error ||:; docker rm -f oci_error ||:' EXIT
++ docker run -d -v /tmp:/workspace --name oci_error centos:6 tail -f /dev/null
+ container_id=1a212e3fb4aa9a2d55260d404a639957d7126a8722a1b767ca8909ab839e590a
+ docker exec --interactive=false 1a212e3fb4aa9a2d55260d404a639957d7126a8722a1b767ca8909ab839e590a date -u
Sun May 15 10:41:11 UTC 2016
++ docker commit --pause=true 1a212e3fb4aa9a2d55260d404a639957d7126a8722a1b767ca8909ab839e590a oci_error
+ image_id=sha256:ae6002376259044f68f1b9e1826523aa7de5ee1499c5d8c7643e16f85be88a59
+ docker exec --interactive=false 1a212e3fb4aa9a2d55260d404a639957d7126a8722a1b767ca8909ab839e590a date -u
rpc error: code = 2 desc = "oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory"+ docker rmi oci_error
Untagged: oci_error:latest
Deleted: sha256:ae6002376259044f68f1b9e1826523aa7de5ee1499c5d8c7643e16f85be88a59
Deleted: sha256:46674d91278a2c66c7bdd9c0dd7d7a860657f0f1eddb1f7c7e43b7a792ea434b
+ docker rm -f oci_error
oci_error
```

**Describe the results you expected:**

Commit should not brake container.

**Additional information you deem important (e.g. issue happens only occasionally):**

The issue is persistent.

**Comment by matusmarhefka** [comment]
For me it happens too, but only in case that I do docker-diff before docker-exec
on a running container:
```
$ docker diff CONT
$ docker exec -it CONT bash
rpc error: code = 2 desc = "oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory"
```

Output of docker-info:
```
$ docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 9
Server Version: 1.11.1
Storage Driver: devicemapper
 Pool Name: docker-8:5-1459624-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.285 GB
 Data Space Total: 107.4 GB
 Data Space Available: 18.53 GB
 Metadata Space Used: 2.277 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93 (2015-01-30)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.8-200.fc22.x86_64
Operating System: Fedora 22 (Twenty Two)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.529 GiB
Name: foo
ID: UD4U:DA4K:7XBG:W2DX:KPI4:YJVH:7V5D:IPJC:XV6I:J33M:7FFW:XADQ
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
```

**Comment by runcom** [comment]
do you have daemon logs?

**Comment by matusmarhefka** [comment]
docker daemon log for the error:
```
time="2016-05-16T11:22:33.414906952+02:00" level=error msg="devmapper: Error unmounting device 9f86489c38c4f44bcce1d104835928647a26dc9bb3ba3bb2aa49a21fe9424852: Device is Busy"
time="2016-05-16T11:22:36.789077675+02:00" level=error msg="Error running exec in container: rpc error: code = 2 desc = \"oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory\"\n"
time="2016-05-16T11:22:36.789125853+02:00" level=error msg="Handler for POST /v1.23/exec/bde30fc82820c0215bf003d7eeae960b6d6151b8d76698042c156f290923d052/start returned error: rpc error: code = 2 desc = \"oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory\""
2016/05/16 11:22:36 http: response.WriteHeader on hijacked connection
2016/05/16 11:22:36 http: response.Write on hijacked connection
time="2016-05-16T11:22:36.789372659+02:00" level=error msg="Handler for POST /v1.23/exec/bde30fc82820c0215bf003d7eeae960b6d6151b8d76698042c156f290923d052/resize returned error: rpc error: code = 2 desc = \"containerd: processs not found for container\""
```

**Comment by jkliff** [comment]
I can confirm it seems related to running `docker diff` beforehand. 
An additional impact I note is being unable to issue `docker restart`.

    $ docker restart 3c605706fe8b
    Error response from daemon: Cannot restart container 3c605706fe8b: rpc error: code = 2 desc = "oci runtime error: rootfs (\"/var/lib/docker/devicemapper/mnt/4798f1af51b13d333827894c08c2feca29d24c3012e44ab6b5697f41d63720a5/rootfs\") does not exist"

docker info:

    Containers: 4
     Running: 3
     Paused: 0
     Stopped: 1
    Images: 64
    Server Version: 1.11.1
    Storage Driver: devicemapper
     Pool Name: docker-202:0-1749902-pool
     Pool Blocksize: 65.54 kB
     Base Device Size: 10.74 GB
     Backing Filesystem: ext4
     Data file: /dev/loop0
     Metadata file: /dev/loop1
     Data Space Used: 2.651 GB
     Data Space Total: 107.4 GB
     Data Space Available: 15.74 GB
     Metadata Space Used: 5.898 MB
     Metadata Space Total: 2.147 GB
     Metadata Space Available: 2.142 GB
     Udev Sync Supported: true
     Deferred Removal Enabled: false
     Deferred Deletion Enabled: false
     Deferred Deleted Device Count: 0
     Data loop file: /var/lib/docker/devicemapper/devicemapper/data
     Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
     Library Version: 1.02.77 (2012-10-15)
    Logging Driver: json-file
    Cgroup Driver: cgroupfs
    Plugins:
     Volume: local
     Network: bridge null host
    Kernel Version: 4.5.0-x86_64-linode65
    Operating System: Ubuntu 14.04.4 LTS
    OSType: linux
    Architecture: x86_64
    CPUs: 2
    Total Memory: 1.945 GiB
    Name: outpost.h7r
    ID: JDQC:5IVO:VLSJ:ZL25:Z6SG:4GAO:B3HR:XQ7L:HV3S:WIBD:A4NB:GOYT
    Docker Root Dir: /var/lib/docker
    Debug mode (client): false
    Debug mode (server): false
    Registry: https://index.docker.io/v1/

**Comment by thaJeztah** [comment]
@mlaventure @tonistiigi any ideas?

**Comment by mlaventure** [comment]
This looks to be a devicemapper issue, I can't reproduce on btrfs or aufs.

and with devicemapper I get the following error while trying `docker commit` or `docker diff`:

```
DEBU[0376] devmapper: activateDeviceIfNeeded(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0376] devmapper: activateDeviceIfNeeded(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: UnmountDevice(hash=396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: Unmount(/var/lib/docker/devicemapper/mnt/396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: Unmount done
DEBU[0377] devmapper: deactivateDevice(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: removeDevice START(docker-253:1-392325-396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: removeDevice END(docker-253:1-392325-396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: deactivateDevice END(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init)
DEBU[0377] devmapper: UnmountDevice(hash=396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d-init) END
DEBU[0377] devmapper: UnmountDevice(hash=396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0377] devmapper: Unmount(/var/lib/docker/devicemapper/mnt/396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0377] devmapper: Unmount done
DEBU[0377] devmapper: deactivateDevice(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0377] devmapper: removeDevice START(docker-253:1-392325-396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0397] devmapper: removeDevice END(docker-253:1-392325-396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0397] devmapper: deactivateDevice END(396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d)
DEBU[0397] devmapper: UnmountDevice(hash=396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d) END
ERRO[0397] devmapper: Error unmounting device 396f98878ccf4fb580dd250117b2fd09ac30ac2f6447b7ef175ba62039edcb0d: Device is Busy
```

**Comment by svdmitrij** [comment]
I faced with same error. After service docker restart all work correctly.

**Comment by thaJeztah** [comment]
So, looks definitely like a devicemapper issue. This fails when using devicemapper,
but no issues after switching to overlay;

```
[root@fedora-2gb-ams3-01 ~]# docker run -d -v /tmp:/workspace --name oci_error centos:6 tail -f /dev/null
03604ad159be110cad888318b66e19665dffc1371ad0b270d91b0b676a10e522
[root@fedora-2gb-ams3-01 ~]# docker exec oci_error date -u
Tue May 17 14:03:27 UTC 2016
[root@fedora-2gb-ams3-01 ~]# docker commit --pause=true oci_error oci_error
sha256:b1e04f2436a0180b05a3f68242622dc8748796ec3348d57fb9644030c7938898
[root@fedora-2gb-ams3-01 ~]# docker exec oci_error date -u
Tue May 17 14:03:53 UTC 2016
```

**Comment by thaJeztah** [comment]
ping @rhvgoyal any ideas what influence device mapper could have here?

**Comment by mlaventure** [comment]
This is fixed by https://github.com/docker/docker/pull/22168 which is only in master.

**Comment by thaJeztah** [comment]
Cool, let me close this issue then

**Comment by theartoflogic** [comment]
Is there a workaround for this until v1.11.2 is released with #22168? I've tried restarting the docker service as mentioned above and I still get this error when trying to exec into a container or restart a container.

I can get around not being able to restart a container by simply removing it and re-creating it. But I cannot get around the issue of being able to exec into a container.

**Comment by espritengineer** [comment]
Hello, the issue is appearing also when executing docker exec command on a running container: 

**output of `docker version` :** 
```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64
```
 
 **output of `docker info`** :
 
```
 Containers: 8
 Running: 8
 Paused: 0
 Stopped: 0
Images: 96
Server Version: 1.11.1
Storage Driver: devicemapper
 Pool Name: docker-202:2-197453-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 3.105 GB
 Data Space Total: 107.4 GB
 Data Space Available: 18.02 GB
 Metadata Space Used: 6.382 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.141 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.10.0-327.18.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 11.33 GiB
Name: solutions
ID: IEKO:LOZU:BKUP:XPE4:KJ5C:TSNY:53HK:C5DA:5G57:TDOQ:P44W:HPFK
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Steps to reproduce the issue:**

docker exec -it mysql

**Describe the results you received:**
rpc error: code = 2 desc = "oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory"

**Describe the results you expected:**
Docker lets me enter bash mode to execute some commands on my container

The only workaround I got working is that I have to restart the container and then docker exec can be available.

**Comment by thaJeztah** [comment]
@espritengineer please update to 1.11.2, and see if you still have this. This issue was fixed in 1.11.1 so it's expected that it still results in issues in 1.11.1

**Comment by joune** [comment]
@thaJeztah We have the same issue on an amazon instance (slave worker for our CI with an AMI provided by bamboo); how can I upgrade (or downgrade) the pre-installed docker? (yum install from the amazon repo only knows version 1.11.1)

**Comment by joune** [comment]
@thaJeztah never mind, i found it:

    yum --showduplicates list docker

showed that version 1.9.1 is also available, then 

    yum remove docker
    yum install docker-1.9.1-1.3.amzn1

**Comment by thaJeztah** [comment]
@joune you'd probably need to ask Amazon, because there's no official packages in our repository for Amazon Linux

**Comment by espritengineer** [comment]
@thaJeztah We upgraded to 1.11.2 as you suggested and the problem is no longer appearing. Thank you.

**Comment by thaJeztah** [comment]
thank you, @espritengineer, that's good to hear!

**Comment by EdVinyard** [comment]
In 1.11.1, `docker ps -s` also causes this problem.

**Comment by pablitok** [comment]
Hi, i'm seeing this issue with docker 1.12 after rebooting the host
`Docker version`

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64
```

`docker info`
```
Containers: 7
 Running: 4
 Paused: 0
 Stopped: 3
Images: 502
Server Version: 1.12.0
Storage Driver: devicemapper
 Pool Name: docker-202:1-133252-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 15.89 GB
 Data Space Total: 107.4 GB
 Data Space Available: 6.482 GB
 Metadata Space Used: 37.88 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.11 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null overlay host bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 3.13.0-92-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.337 GiB
Name: ip-10-253-169-151
ID: 2OKP:CNUB:WQQ7:5RQ7:LRWT:F7CN:NW4N:M2ZQ:GSPA:3X6Q:OGVH:V4XD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```
`docker-compose version`
```
docker-compose version 1.7.1, build 0a9ab35
docker-py version: 1.8.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```
I'm running Ubuntu 14.04.4 with kernel 3.13.0 in amazon ec2.
After rebooting, the containers are automatically started with  `restart: unless-stopped` in the compose file; this is when the issue appears, if I stop the container and manually run it, the problem dissapears.

**Comment by thaJeztah** [comment]
@pablitok could you open a new issue, it could have a different cause, so better to start a "fresh" issue

**Comment by rhuddleston** [comment]
I'm on docker version 1.11.2 and seeing this issue:

# docker exec -it 349aba944bf7 /bin/bash
rpc error: code = 2 desc = "oci runtime error: exec failed: fork/exec /proc/self/exe: no such file or directory"

docker version:

```Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 22:00:43 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 22:00:43 2016
 OS/Arch:      linux/amd64```

docker info 

```Containers: 7
 Running: 7
 Paused: 0
 Stopped: 0
Images: 7
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 2.448 GB
 Data Space Total: 22.44 GB
 Data Space Available: 19.99 GB
 Metadata Space Used: 1.192 MB
 Metadata Space Total: 234.9 MB
 Metadata Space Available: 233.7 MB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.110 (2015-10-30)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.0-31-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.952 GiB
Name: ecs-177471654.stage.centricient.corp
ID: T5V4:FP64:4D4C:XGCT:VMZG:SKSD:PQ44:WR44:DIHA:YU2K:KVDW:WM2P
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support```

**Comment by Spindel** [comment]
We get this regularly on **1.12.0** as well:
```rpc error: code = 2 desc = oci runtime error: exec failed: process_linux.go:75: starting setns process caused "fork/exec /proc/self/exe: no such file or directory```

```
[spider@front ~]$ docker info
Containers: 18
 Running: 18
 Paused: 0
 Stopped: 0
Images: 20
Server Version: 1.12.0
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 29.1 GB
 Data Space Total: 32.64 GB
 Data Space Available: 3.532 GB
 Metadata Space Used: 33.71 MB
 Metadata Space Total: 339.7 MB
 Metadata Space Available: 306 MB
 Thin Pool Minimum Free Space: 3.263 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host overlay bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.28.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 2.939 GiB
Name: CENSORED
ID: UTRC:LXPI:RCDP:365H:26ZB:QS5H:3TM5:F4MO:3XN7:277X:7YOB:4EJQ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
```

**Comment by thaJeztah** [comment]
@Spindel could you please continue in https://github.com/docker/docker/issues/25381, which was opened for 1.12 (it may be a different issue)

I'm locking this issue so that we can concentrate the discussion in a single place; if you're still encountering this on 1.12, please subscribe to https://github.com/docker/docker/issues/25381
